anti_patterns/deny-warnings.md

```diff
@@ -0,0 +1,35 @@
+# `#![deny(warnings)]`
+
+## Description
+
+A well-intentioned crate author wants to ensure their code builds without
+warnings. So they annotate their crate root with
+
+## Example
+
+```rust
+#![deny(warnings)]
+
+// All is well.
+```
+
+## Drawbacks
+
+Until, say, the language changes, and as part of the transition, a lint is
+introduced that `warn`s, and Boom! The code no longer builds. Or someone tries
```

Comment:
I would phrase this in terms of the Rust language's back compat guarantees - we guarantee (mostly) that we won't break your code with new releases, however, that guarantee does not cover warnings, and thus by using `deny(warnings)` you are opting out of backwards compatibility for your crate.